components/base/sensorcontrolled/sensorcontrolled.go

```diff
@@ -158,11 +171,47 @@ func (sb *sensorBase) Reconfigure(ctx context.Context, deps resource.Dependencie
 		return errors.Wrapf(err, "no base named (%s)", newConf.Base)
 	}
 
-	if sb.velocities != nil && useControlLoop {
-		sb.controlLoopConfig = sb.createControlLoopConfig()
-		if err := sb.setupControlLoops(); err != nil {
-			return err
+	if sb.velocities != nil && len(newConf.ControlParameters) != 0 {
+		// assign linear and angular PID correctly based on the given type
+		var linear, angular basePIDConfig
+		for _, c := range newConf.ControlParameters {
+			if c.Type == typeLinVel {
+				linear = c
+			} else if c.Type == typeAngVel {
+				angular = c
+			}
 		}
+
+		sb.mu.Unlock()
```

Comment:
Why are we unlocking here?

Reply:
since the tuning process needs to run the motors, this lock prevents the tuning process from starting, so it needs to unlock before anything to do with the tuning, and then relock at the end of the function since there is a defer unlock fro the top of this function. kind of weird I'll add some comments